sklearn/ensemble/tests/test_bagging.py

```diff
@@ -907,3 +907,30 @@ def fit(self, X, y):
 
     assert_array_equal(clf.estimators_[0]._sample_indices,
                        clf.estimators_samples_[0])
+
+
+class ConstantClassifier(ClassifierMixin, BaseEstimator):
+    """Classifier that predict a constant value and that does not implement
+    `predict_proba`."""
+
+    def __init__(self, constant=0):
+        self.constant = constant
+
+    def fit(self, X, y, sample_weight=None):
+        self.classes_ = np.unique(y)
+        return self
+
+    def predict(self, X):
+        return np.full(X.shape[0], self.constant)
```

Comment:
This example is kind of strange because the return of `predict` is a bool, while `classes_` is `[0, 1]`. `BaggingClassifer` will encode `y` before passing it to its estimators in fit when it calls `_validate_y`

https://github.com/scikit-learn/scikit-learn/blob/9183486463c1df3b5b3c7e2357e17533bdd36573/sklearn/ensemble/_bagging.py#L292

https://github.com/scikit-learn/scikit-learn/blob/9183486463c1df3b5b3c7e2357e17533bdd36573/sklearn/ensemble/_bagging.py#L651-L657

Reply:
Hm, now I see this mostly arose due to the way I coded up the estimator. It's been quite a while since the original PR so I don't fully remember the context. IRC, I was expecting BaggingClassifier to just take the mode across axis 1 of whatever the ensemble members returned. But now I see that it requires the estimators to handle class labels like other sklearn estimators. Perhaps the way I was thinking originally might eliminate the need for this (unstated?) requirement, but it's probably fine as is.